File: diaspora-stream-api/docs/_code/producer.py
import sys
from diaspora_stream.api import Driver


def produce(root_path: str, topic_name: str):
    driver_options = {
        "root_path": root_path,
    }
    driver = Driver(backend="files", options=driver_options)
    topic = driver.open_topic(topic_name)
    producer = topic.producer()

    for i in range(0, 100):
        future = producer.push(
            metadata={"x": i*42, "name": "bob"},
            data=bytes())
        event_id = future.wait()


if __name__ == "__main__":

    if len(sys.argv) != 3:
        print(f"Usage: {sys.argv[0]} <root_path> <topic_name>")
    root_path = sys.argv[1]
    topic_name = sys.argv[2]
    produce(root_path, topic_name)

File: diaspora-stream-api/docs/_code/producer.cpp
#include <diaspora/Driver.hpp>
#include <diaspora/TopicHandle.hpp>
#include <iostream>

int main(int argc, char** argv) {

    if(argc != 3) {
        std::cerr << "Usage: " << argv[0] << " <root_path> <topic>" << std::endl;
        return -1;
    }

    auto root_path = argv[1];
    auto topic_name = argv[2];

    try {

        diaspora::Metadata driver_options;
        driver_options.json()["root_path"] = root_path;

        diaspora::Driver driver = diaspora::Driver::New("files", driver_options);

        diaspora::TopicHandle topic = driver.openTopic(topic_name);

        diaspora::Producer producer = topic.producer();
        for(size_t i = 0; i < 100; ++i) {
            auto future = producer.push(
                diaspora::Metadata{R"({"x":42,"name":"bob"})"},
                diaspora::DataView{}
            );
            // auto event_id = future.wait(-1).value();
        }
        producer.flush().wait(-1);

    } catch(const diaspora::Exception& ex) {
        std::cerr << ex.what() << std::endl;
    }
    return 0;
}
